sycl/source/detail/os_util.cpp

```diff
@@ -56,6 +56,8 @@ struct ModuleInfo {
   const char *Name;     // out
 };
 
+constexpr OSModuleHandle OSUtil::ExeModuleHandle;
```

Comment:
It provides a definition for the static member variable.

Reply:
It looks more like a declaration. This variable is defined in os_util.hpp which is included in the file. So can't understand why this line is necessary.

Reply:
It is in fact a definition, while the line in os_util.hpp is a declaration (until C++17). From [https://en.cppreference.com/w/cpp/language/definition](url):

> Every declaration is a definition, except for the following:
> ...
> Declaration of a non-inline (since C++17) static data member inside a class definition

The line in os_util.hpp does provide an initializer as constexpr static member variable declarations are required to do, but it's still just a declaration. The definition has to be provided in one of the translation units in order to comply with ODR.

Reply:
Ok.